### PR TITLE
(RSP-1825) Adding basic docs for overlay aria hooks

### PR DESCRIPTION
**Pull request by LFDanLu**
Closes https://jira.corp.adobe.com/browse/RSP-1825

## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [Issue](https://github.com/adobe-private/react-spectrum-v3/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

### CODE REVIEW

File: packages/@react-aria/overlays/docs/useModal.mdx [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+---
+
+# useModal
+
+<p>{docs.exports.useModal.description}</p>
+
+<HeaderInfo
+  packageData={packageData}
+  componentNames={['useModal']} />
+
```
Comment (LFDanLu) [inline packages/@react-aria/overlays/docs/useModal.mdx:29]: omitted a intro section here cuz the description felt good enough (was struggling to add something meaningful instead of just regurgitating the description a second time).

File: packages/@react-aria/overlays/docs/useModal.mdx [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+## Example
```
Comment (snowystinger) [inline packages/@react-aria/overlays/docs/useModal.mdx:30]: missing an API section?

Comment (LFDanLu) [reply packages/@react-aria/overlays/docs/useModal.mdx:30]: useModal actually doesn't have any args and doesn't return anything so the FunctionAPI component doesn't render, lemme see if I can change it so it shows that

Comment (snowystinger) [reply packages/@react-aria/overlays/docs/useModal.mdx:30]: haha, great, yeah, we should probably still show that if possible

Comment (LFDanLu) [reply packages/@react-aria/overlays/docs/useModal.mdx:30]: nvm I'm dumb, forgot to import it zzzzzz

File: packages/@react-aria/overlays/docs/useModal.mdx [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+import packageData from '../package.json';
+
+---
+category: Overlays
+keywords: [dialog, modal, tray, popover, aria]
+---
+
+# useModal
+
+<p>{docs.exports.useModal.description}</p>
```
Comment (ktabors) [inline packages/@react-aria/overlays/docs/useModal.mdx:24]: Why is this description showing the `'s instead of formatting the word?

Comment (LFDanLu) [reply packages/@react-aria/overlays/docs/useModal.mdx:24]: Not sure, I dunno how much formatting happens on the description. Mind if I handle that in a followup?

Comment (devongovett) [reply packages/@react-aria/overlays/docs/useModal.mdx:24]: mmm yeah we should probably wrap this in a markdown renderer. followup is fine

File: packages/@react-aria/overlays/docs/useOverlayPosition.mdx [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+<p>{docs.exports.useOverlayPosition.description}</p>
+
+<HeaderInfo
+  packageData={packageData}
+  componentNames={['useOverlayPosition']} />
+
+## API
+
+<FunctionAPI function={docs.exports.useOverlayPosition} links={docs.links} />
```
Comment (devongovett) [inline packages/@react-aria/overlays/docs/useOverlayPosition.mdx:32]: Seems like the options/result types are missing descriptions for the properties. Could be a separate pr if you want

Comment (LFDanLu) [reply packages/@react-aria/overlays/docs/useOverlayPosition.mdx:32]: yep, should be handled in https://github.com/adobe-private/react-spectrum-v3/pull/634/files
